### PR TITLE
ANDROID: EAS fixes

### PR DESCRIPTION
**Pull request by ccawley2011**

### CODE REVIEW

File: audio/softsynth/eas.cpp [hunk 1]
```diff
@@ -280,27 +270,10 @@ int MidiDriver_EAS::open() {
...
-		EASFile f;
-		memset(&f, 0, sizeof(EASFile));
-		f.path = dls.c_str();
-
-		res = _loadDLSFunc(_EASHandle, 0, &f);
-		if (res)
-			warning("error loading DLS file '%s': %d", dls.c_str(), res);
-		else
-			debug("DLS file loaded");
-	}
```
Comment (bluegr) [inline audio/softsynth/eas.cpp]: So, why has this been removed? And is there any other way of loading sound fonts?

File: audio/softsynth/eas.cpp [hunk 2]
```diff
@@ -84,37 +84,30 @@ class MidiDriver_EAS : public MidiDriver_MPU401, Audio::AudioStream {
...
-		int32 bufSize;
-		uint32 filter;
-		uint32 timeStamp;
+		unsigned long version;
+		unsigned debug;
+		long voices;
+		long channels;
+		long rate;
+		long bufSize;
+		unsigned filter;
```
Comment (bluegr) [inline audio/softsynth/eas.cpp]: Where did you find the types? And can you please use ScummVM types?

Comment (ccawley2011) [reply audio/softsynth/eas.cpp]: The types come from here: https://android.googlesource.com/platform/external/sonivox/+/refs/heads/master/arm-wt-22k/host_src/eas_types.h#110

The structure is a copy of `S_EAS_LIB_CONFIG`, and is passed to the dynamically loaded EAS functions. As such, the structure should be using the types that EAS is expecting.

File: audio/softsynth/eas.cpp (hunk 1 above)
Comment (ccawley2011) [reply audio/softsynth/eas.cpp]: I already explained why in the relevant commit message. As far as I know, there isn't any other way of loading soundfonts with EAS.

Comment (bluegr) [reply audio/softsynth/eas.cpp]: Perhaps we could use the Android version for this? This change was probably in the codebase in Android version 5.1 and later versions:
https://en.m.wikipedia.org/wiki/Android_version_history

Comment (lephilousophe) [reply audio/softsynth/eas.cpp]: Maybe a configure check is more robust than just relying on Android version?

**Comment by antoniou79** [comment]
Could we look into using the following midi driver to address this?
https://github.com/billthefarmer/mididriver

 The Sonivox library is now built as a static library and linked with the midi library. This should resolve problems with various android versions.

It seems related work, and it's actively maintained.
Sorry, in advance, if it's actually not relevant.

**Comment by sev-** [comment]
What is the status of this PR?

**Comment by lephilousophe** [comment]
EAS is not working anymore on latest Android versions (since 7.0 it seems): it was not a public NDK API and access to it is now barred.
In addition, commit aosp-mirror/platform_external_sonivox@af41595537b044618234fe7dd9ebfcc652de1576 removed the ability to load DLS collections (and it even didn't seem to work when tested in https://github.com/billthefarmer/mididriver/issues/21).
There is this fork [pedrolcl/sonivox](https://github.com/pedrolcl/sonivox) which reverts things and which is agnostic of Android but does this worth the pain?

**Comment by ccawley2011** [comment]
> There is this fork [pedrolcl/sonivox](https://github.com/pedrolcl/sonivox) which reverts things and which is agnostic of Android but does this worth the pain?

This might actually be quite interesting, since FluidSynth can be quite slow on some platforms. I'll look into this in the future.

**Comment by bluegr** [comment]
Closing this in favor of #4712

**Pull request closed without merging**